File: sdk-dotnet/LittleHorse.Sdk/Exceptions/LHSerdeException.cs
namespace LittleHorse.Sdk.Exceptions
{
    ///<summary>
    /// Maps Exception that contains serialization or deserialization of objects in LH context.
    /// </summary>
    public class LHSerdeException : Exception
    {
        ///<summary>
        /// Empty Constructor of the Exception which represents a problem with
        /// serialization or deserialization of objects in LH context.
        /// </summary>
        public LHSerdeException() : base() { }

        ///<summary>
        /// Constructor of the Exception which represents a problem with
        /// serialization or deserialization of objects in LH context.
        /// </summary>
        /// <param name="message"> A custom message.</param>
        public LHSerdeException(string message) : base(message) { }

        ///<summary>
        /// Constructor of the Exception which represents a problem with
        /// serialization or deserialization of objects in LH context.
        /// </summary>
        /// <param name="message"> A custom message.</param>
        /// <param name="innerException">This is another exception that caused this one.</param>
        public LHSerdeException(string message, Exception innerException) : base(message, innerException) { }

    }
}

File: sdk-java/src/main/java/io/littlehorse/sdk/common/exception/LHSerdeException.java
package io.littlehorse.sdk.common.exception;

public class LHSerdeException extends RuntimeException {

    private Throwable exn;
    private String message;

    public LHSerdeException(Throwable exn, String message) {
        this.exn = exn;
        this.message = message;
    }

    public LHSerdeException(String message) {
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    public Throwable getParent() {
        return exn;
    }
}
